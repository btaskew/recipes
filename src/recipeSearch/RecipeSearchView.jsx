import React from 'react';
import PropTypes from 'prop-types';

import Page from 'components/Page';
import InputForm from 'components/InputForm';
import RecipesList from 'components/RecipesList';

function RecipeSearchView(props) {
    return (
        <Page heading="Search by ingredient" loading={props.loading}>
            <InputForm
                submitForm={props.handleSubmit}
                buttonText="Search"
                inputText="Enter ingredient"
            />

            {!props.loading && <RecipesList styleName="margin-top" recipes={props.recipes} />}
        </Page>
    );
}

RecipeSearchView.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    recipes: PropTypes.array
};
RecipeSearchView.defaultProps = {
    recipes: []
};

export default RecipeSearchView;
