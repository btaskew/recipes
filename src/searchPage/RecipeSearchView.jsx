import React from 'react';
import PropTypes from 'prop-types';

import {Alert} from 'react-bootstrap';
import Page from 'components/Page';
import InputForm from 'components/InputForm';
import RecipesList from 'components/RecipesList';

function RecipeSearchView(props) {
    return (
        <Page heading="Search by ingredient" loading={props.loading}>
            <InputForm
                submitForm={props.handleSubmit}
                buttonText="Search"
                inputText="Enter ingredient"
            />

            {props.noResult && (
                <Alert bsStyle="info" className="marginTop">
                    No recipes found for that ingredient
                </Alert>
            )}

            {props.recipes && <RecipesList styleName="marginTop" recipes={props.recipes} />}
        </Page>
    );
}

RecipeSearchView.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    noResult: PropTypes.bool.isRequired,
    recipes: PropTypes.array
};
RecipeSearchView.defaultProps = {
    recipes: []
};

export default RecipeSearchView;
