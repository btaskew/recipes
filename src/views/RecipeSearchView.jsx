import React from 'react';

import {Alert} from 'react-bootstrap';
import Page from 'components/utils/Page';
import InputForm from 'components/utils/InputForm';
import RecipesList from 'components/RecipesList';

function RecipeSearchView(props) {
    return (
        <Page heading="Search by ingredient" loading={props.loading}>
            <InputForm
                submitForm={props.handleSubmit}
                buttonText="Search"
                inputText="Enter ingredient"
            />

            {props.noResult && (
                <Alert bsStyle="info">
                    No recipes found for that ingredient
                </Alert>
            )}

            {props.error && (
                <Alert bsStyle="danger">{props.errorMessage}</Alert>
            )}

            {props.recipes && (
                <RecipesList styleName="topPadding" recipes={props.recipes} />
            )}
        </Page>
    );
}

export default RecipeSearchView;
