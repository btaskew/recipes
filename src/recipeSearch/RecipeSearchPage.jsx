import React from 'react';
import RecipeSearch from 'containers/RecipeSearch';
import RecipeSearchView from 'recipeSearch/RecipeSearchView';

function RecipeSearchPage() {
    return (
        <RecipeSearch
            render={(state, recipeSearch) => (
                <RecipeSearchView {...state} handleSubmit={recipeSearch} />
            )}
        />
    );
}

export default RecipeSearchPage;
