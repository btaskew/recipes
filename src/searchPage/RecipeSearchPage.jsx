import React from 'react';
import RecipeSearch from 'containers/RecipeSearch';
import RecipeSearchView from 'searchPage/RecipeSearchView';

function RecipeSearchPage() {
    return (
        <RecipeSearch
            render={(state, recipeSearch) => (
                <RecipeSearchView {...state} handleSubmit={recipeSearch} />
            )}
        />
    );
}

export default RecipeSearchPage;
