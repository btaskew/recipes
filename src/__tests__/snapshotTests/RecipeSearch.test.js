import React from 'react';
import renderer from 'react-test-renderer';

import RecipeSearchView from 'recipeSearch/RecipeSearchView';
import RecipeSearchPage from 'recipeSearch/RecipeSearchPage';
import {MemoryRouter} from 'react-router';

describe('Recipe search page', () => {
    it('renders correctly', () => {
        const searchPage = renderer
            .create(
                <MemoryRouter>
                    <RecipeSearchPage result={false}  />
                </MemoryRouter>
            )
            .toJSON();
        expect(searchPage).toMatchSnapshot();
    });
});

describe('Recipe search view', () => {
    it('renders loading correctly', () => {
        const searchPage = renderer
            .create(
                <MemoryRouter>
                    <RecipeSearchView loading={true} result={false} handleSubmit={() => {}} />
                </MemoryRouter>
            )
            .toJSON();
        expect(searchPage).toMatchSnapshot();
    });

    it('renders with no result correctly', () => {
        const searchPage = renderer
            .create(
                <MemoryRouter>
                    <RecipeSearchView loading={true} recipes={null} result={true} handleSubmit={() => {}} />
                </MemoryRouter>
            )
            .toJSON();
        expect(searchPage).toMatchSnapshot();
    });

    it('renders with results correctly', () => {
        const testRecipes = [
            {
                $loki: 1,
                name: 'Recipe 1'
            },
            {
                $loki: 2,
                name: 'Recipe 2'
            }
        ];

        const searchPage = renderer
            .create(
                <MemoryRouter>
                    <RecipeSearchView loading={false} recipes={testRecipes} result={true} handleSubmit={() => {}} />
                </MemoryRouter>
            )
            .toJSON();
        expect(searchPage).toMatchSnapshot();
    });
});
