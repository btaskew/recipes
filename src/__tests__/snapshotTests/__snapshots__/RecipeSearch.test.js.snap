// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe search page renders correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Search by ingredient
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="flex"
    >
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Enter ingredient"
        type="text"
        value=""
      />
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>,
]
`;

exports[`Recipe search view renders loading correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Search by ingredient
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      Loading...
    </div>
  </div>,
]
`;

exports[`Recipe search view renders with no result correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Search by ingredient
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      Loading...
    </div>
  </div>,
]
`;

exports[`Recipe search view renders with results correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Search by ingredient
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="flex"
    >
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Enter ingredient"
        type="text"
        value=""
      />
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
    <div
      className="margin-top list-group"
    >
      <a
        className="list-group-item"
        href="/recipe/1"
        onClick={[Function]}
        style={undefined}
      >
        Recipe 1
      </a>
      <a
        className="list-group-item"
        href="/recipe/2"
        onClick={[Function]}
        style={undefined}
      >
        Recipe 2
      </a>
    </div>
  </div>,
]
`;
